### PR TITLE
Adding hasTagName assertion

### PR DESCRIPTION
**Pull request by scalvert**
Implements #493.

### CODE REVIEW

**Comment by scalvert** [comment]
I believe I need to add an entry in the ember tests.

File: lib/assertions.ts [hunk 1]
```diff
@@ -955,6 +956,57 @@ export default class DOMAssertions {
...
+
+    if (typeof tagName !== 'string') {
+      throw new TypeError(`You must pass a string to "hasTag". You passed ${tagName}.`);
+    }
+
+    if (!TAG_NAMES.includes(tagName.toLowerCase())) {
+      throw new Error(
+        `The tagName '${tagName}' is not a valid HTML tag. You must provide a valid HTML tag`
+      );
+    }
```
Comment (Turbo87) [inline lib/assertions.ts]: this seems a bit too restrictive to me. at some point new tag names might be introduced or people will use webcomponents that use custom tag names.

File: lib/assertions.ts [hunk 2]
```diff
@@ -955,6 +956,57 @@ export default class DOMAssertions {
...
+
+    actual = element.tagName.toLowerCase();
+    expected = tagName;
```
Comment (Turbo87) [inline lib/assertions.ts]: we should probably apply `toLowerCase()` to the passed in `tagName` too 🤔

File: lib/assertions.ts (hunk 1 above)
Comment (scalvert) [reply lib/assertions.ts]: Ya this was a toss up, and to be fair I’m happy to remove it. If the caller passes an invalid HTML tag the test will fail anyway, so maybe that’s enough?

File: lib/assertions.ts (hunk 2 above)
Comment (scalvert) [reply lib/assertions.ts]: Agree. I'll clean it up a bit.

File: lib/assertions.ts [hunk 3]
```diff
@@ -955,6 +955,51 @@ export default class DOMAssertions {
...
+   * @param {string?} message
+   *
+   * @example
+   * // <h1 id="title">
+   * //   Title
+   * // </h1>
+   *
+   * assert.dom('#title').hasTag('h1');
+   */
+  hasTag(tagName: string, message?: string) {
```
Comment (Turbo87) [inline lib/assertions.ts]: I think I would prefer `hasTagName()` 🤔 

thoughts?

Comment (scalvert) [reply lib/assertions.ts]: Ha, I quite literally thought the same thing. I'll make the change now.

**Comment by scalvert** [comment]
@Turbo87 I forget how to make the ember tests work. I added an assertion, but `hasTagName` doesn't appear to be available in the version of `qunit-dom` the tests are using. I remember having this issue before, but can't remember how I resolved :)

**Comment by scalvert** [comment]
I believe I figured it out. Added ember test.

**Comment by scalvert** [comment]
I think this PR should be closely followed by `doesNotHaveTagName`

File: tests/acceptance/qunit-dom-test.js [hunk 4]
```diff
@@ -18,6 +18,7 @@ module('Acceptance | qunit-dom', function(hooks) {
     assert.dom('#title').exists();
     assert.dom('#subtitle').doesNotExist();
     assert.dom('#title').hasText('Welcome to Ember');
+    assert.dom('#title').hasTagName('h2');
```
Comment (Turbo87) [inline tests/acceptance/qunit-dom-test.js]: FWIW we don't have to add all the assertions here. this is just meant as a smoke test that the qunit-dom integration works correctly. the assertions itself are tested enough by the Jest test suite.

**Pull request merged**